protocol_grpc.go

```diff
@@ -361,6 +361,7 @@ func (cc *grpcClientConn) Receive(msg any) error {
 		// means that we're _not_ getting a message. For users to realize that
 		// the stream has ended, Receive must return an error.
 		serverErr.meta = cc.responseHeader.Clone()
+		serverErr.isServerErr = true
```

Comment:
This misses the code path in line 347. Rather than setting `isServerErr` up at this level, we should do it in `grpcErrorFromTrailer` - we're much closer to the wire there.